README.md

```diff
@@ -0,0 +1,104 @@
+| 4                  | Intro to OOP                                                | [lecture-6](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-6)                                                  | 2025-10-05    | No             |
+| 5                  | Core OOP                                                    | [lecture-7](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-7)                                                  | 2025-10-10    | No             |
+| 6                  | Exception Handling                                          | [lecture-8](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-8)                                                  | 2025-11-11    | No              |
+| 7                  | Collections Framework                                       | [lecture-9](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-9), [lecture-10](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-10) | 2025-11-12, 2025-11-17    | Yes (Partial)             |
+| 8                  | Generics                                                    | [lecture-10](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-10), [lecture-11](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-11) | 2025-11-18    | No             |
+| 9                  | File I/O and Concurrency                                    | [lecture-11](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-11), [lecture-12](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-12) | 2025-11-18, 2025-11-24    | No              |
+| 10                 | Advanced Java (Records, Sealed Classes, Annotations, Enums) | [lecture-13](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-13), [lecture-14](https://github.com/Nikhilnair48/guvi-java-gen-ai/tree/main/lecture-14) | 2025-11-25, 2025-11-31    | Yes (Partial)             |
+
+## Learning Experience
+
+Each lecture connects a conceptual idea to its implementation. The process follows a steady rhythm:
+
+1. Introduce a problem that mirrors something real.  
+2. Explore how Java provides tools to address that problem.  
+3. Build a working example, test it, and refine it.  
+4. Reflect briefly on why this approach works better than alternatives.
+
+| Component     | Purpose                                                      |
+| ------------- | ------------------------------------------------------------ |
+| Lecture Notes | Explain the concept and its role in the larger system.       |
+| Live Coding   | Demonstrate the step-by-step reasoning through code.         |
+| Exercises     | Strengthen understanding by applying concepts independently. |
+
+The goal is not to rush through content but to make learners comfortable reading, reasoning about, and improving code.
+
+## Setup Instructions
+
+**Prerequisites:**
```

Comment:
Should eventually hyperlink these. Download link for Java, IntelliJ, etc. Exists in lecture 1 or 2